### PR TITLE
Fix tests on Windows

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Issue: https://github.com/storybookjs/storybook/issues/19763 Some tests were commented out on windows because they were failing

## What I did

Fixed some tests on Windows machines

## How to test

- Run `yarn test` on a Windows machine

### CODE REVIEW

File: code/lib/builder-manager/src/utils/files.test.ts
Comment (tmeasday) [inline code/lib/builder-manager/src/utils/files.test.ts:27]: I wonder if it would be better to switch inside the test, e.g. 

```js
    const addonsDir = os === 'win32' ? 'C:\\Users\\username\\Projects\\projectname\\storybook' : '/Users/username/Projects/projectname/storybook';
```

Comment (tmeasday) [reply code/lib/builder-manager/src/utils/files.test.ts:27]: Or even a pattern like:

```js
function maybeWindows(path) {
  if (os !=== 'win32') return path;

  return path.startsWith('/') ? `C:\\${slash(path)}` : slash(path);
}

const addonsDir = maybeWindows('/Users/username/Projects/projectname/storybook')
```

File: code/lib/core-server/src/utils/StoryIndexGenerator.ts
Comment (tmeasday) [inline code/lib/core-server/src/utils/StoryIndexGenerator.ts:504]: OK, this makes sense, I guess we drop this code?

Comment (valentinpalkovic) [reply code/lib/core-server/src/utils/StoryIndexGenerator.ts:504]: Sure! Just wanted to know your opinion about this first.

File: code/lib/builder-manager/src/utils/files.test.ts
Comment (valentinpalkovic) [reply code/lib/builder-manager/src/utils/files.test.ts:27]: It's not just the addons dir, which changes, but also the path in line 31 and the assertion in line 37. I would concentrate the windows specific changes into an extra test rather inlining incoherent in three different places. Though, I don't have a strong opinion about that :)

Comment (tmeasday) [reply code/lib/builder-manager/src/utils/files.test.ts:27]: I guess I'm just thinking the tests could drift apart if we don't do it inline.

Comment (valentinpalkovic) [reply code/lib/builder-manager/src/utils/files.test.ts]: Alright! I have inlined the condition.

File: code/lib/builder-manager/src/utils/files.test.ts
Comment (yannbf) [inline code/lib/builder-manager/src/utils/files.test.ts:22]: I wonder if this would be better as "toEqual" rather than inlineSnapshot, as jest normally overwrites inlineSnapshots if accepted, and therefore the ternary might go away

Comment (kasperpeulen) [reply code/lib/builder-manager/src/utils/files.test.ts:22]: Or maybe:

```ts
if (isWindows) {
  expect(location).toMatchInlineSnapshot(`"C:\\\\Users\\\\username\\\\Projects\\\\projectname\\\\storybook\\\\node_modules\\\\@storybook\\\\addon-x+y\\\\dist\\\\manager.mjs"`);
} else {
  expect(location).toMatchInlineSnapshot(`"/Users/username/Projects/projectname/storybook/node_modules/@storybook/addon-x+y/dist/manager.mjs"`);
}
```

Comment (valentinpalkovic) [reply code/lib/builder-manager/src/utils/files.test.ts:22]: @kasperpeulen  -> expect shouldn't be called conditionally. Of course I can just disable the check for this line [eslint[jest/no-conditional-expect](https://github.com/jest-community/eslint-plugin-jest/blob/v24.7.0/docs/rules/no-conditional-expect.md)]

Comment (kasperpeulen) [reply code/lib/builder-manager/src/utils/files.test.ts:22]: I see, to equal may be better than yes

**Review by ndelangen — APPROVED** [review]

File: code/lib/cli/src/helpers.ts
Comment (yannbf) [inline code/lib/cli/src/helpers.ts:200]: When do we need to use posix.join instead of path.join?

File: code/jest.config.js
Comment (yannbf) [inline code/jest.config.js:3]: The meaning of this comment changed. Can you update it? E.g. Explain what challenge blocks us from fixing the other tests

**Review by tmeasday — APPROVED** [review]
LGTM, thanks @valentinpalkovic

File: code/lib/cli/src/helpers.ts
Comment (valentinpalkovic) [reply code/lib/cli/src/helpers.ts]: posix.join should be used, if you want to use the path seperator of posix systems, independently, on which system you are. 

In this case though it doesn't make much sense, because all the paths are used to be passed to fse.copy, which can handle path separators on both systems. My first assumption was, that these paths are used to resolve node modules with node's resolution algorithm. Therefore it had made sense to use forward slash, because we don't require packages with a backslash. This assumption was wrong and therefore I have reverted the these changes.

**Comment by IanVS** [comment]
I see this has a few approvals, and I'm waiting on it to be merged for one of my PRs, so I'm going to go ahead and pull the trigger.  Great work on this, @valentinpalkovic!

**Pull request merged**
